subsys/bluetooth/audio/tbs_client.c

```diff
@@ -214,8 +230,9 @@ static void call_cp_callback_handler(struct bt_conn *conn, int err,
 		cp_cb(conn, err, index, call_index);
 	}
 }
+#endif /* defined(CONFIG_BT_TBS_CLIENT_OPTIONAL_CP_OPCODES) */
 
-static const char *parse_string_value(const void *data, uint16_t length,
+static const __attribute__((unused)) char *parse_string_value(const void *data, uint16_t length,
```

Comment:
I'd rather have that we guard this function by whatever we need than using the `__attribute__((unused))`. IIRC `__attribute__((unused))` is also a GCC attribute, and may not work well with other compilers.

Reply:
@fredrikdanebjer Will you look into this one too?

Reply:
I don't think suppressing the warning is the right way. 

I don't mind guarding this with several Kconfigs.

There are alternatives if you don't like the guarding; we can remove the `static` part of the function and make it `inline` and put it in the `tbs_internal.h` file, or simply make the function non-static and keep as is, but still add the prototype to `tbs_internal.h`

We may need/want to fix the function as well, as 1) it doesn't need to allocate a `static` buffer (of course the alternative is that we allocate quite large buffers on the stack), and 2) that it also uses `CONFIG_BT_TBS_MAX_URI_LENGTH`, which may be as low as 4, which is not ideal for parsing e.g. the Provider Name.

Reply:
The quick fix will of course just be adding the Kconfig guards :)

Reply:
I did the second option here.

There was also another point that was brought up in regards to this function and the `CONFIG_BT_TBS_MAX_URI_LENGTH`, since the spec. explicitly says it has to be UTF-8, shouldn't we take height or check for the non-ASCII parts too?
E.g. if we get an 'å' at the end we just have one octet left, maybe we should throw it away?